### ISSUE TITLE
Question : setup SSL

### ISSUE THREAD

**Issue by Zciwobukaj** [firefly-iii/firefly-iii#1212]
Hello,

I try run Firefly 4.7.0 with SSL : 
```
docker run -d \
-v firefly_iii_export:/var/www/firefly-iii/storage/export \
-v firefly_iii_upload:/var/www/firefly-iii/storage/upload \
-p 32782:80 -p 32783:443 \
-e FF_APP_ENV=local \
-e FF_APP_KEY=S0m3R@nd0mString0f32Ch@rsEx@ct1y \
-e FF_DB_HOST=10.11.12.13 \
-e FF_DB_NAME=FIREFLY_DB \
-e FF_DB_USER=FIREFLY_USER \
-e FF_DB_PASSWORD=FIREFLY_PASSWORD \
-e APP_URL=https://my.domain \
-e APP_FORCE_SSL=true \
-e APP_FORCE_ROOT=https://my.domain \
-e TRUSTED_PROXIES=**
jc5x/firefly-iii:latest
```

Port 32782 (HTTP) works fine.
Port 32783 (HTTPS) sends some access denied error.

I would like not use any proxy (is there any reason that this is needed ?). Is it possible to do so ?
And is this the reason why i have this error ?

Thanks for your help

**Comment by setevoy2** [comment]
Firefly itself has no SSL included (afaik) and it will not listen 443 port with SSL certificate configured. That's why you have to setup some reverse-proxy like NGINX which will handle SSL requests for you and will proxy requests to a Firefly container.

PS I have NGINX with SSL for my setup and Docker Compose with FF-III in an `upstream`.

I may be wrong, @JC5 knows better :-)

**Comment by Zciwobukaj** [comment]
@setevoy2 : thanks for your answer
I tried configure jwilder/nginx-proxy with JrCs/docker-letsencrypt-nginx-proxy-companion to enable SSL.
Is this how you made this ? When I tried, I couldn't neither register any certificate nor access to port 443...
Truly this would help me if you can guide me.

Thanks

**Comment by JC5** [comment]
I'm not sure how this can be fixed @Zciwobukaj, it's a little out of scope for me. But the docker image is not prepped for TLS. It has no certificates or config for TLS. You can open ports of course but it won't make it happen I'm afraid.

How to set up a reverse proxy can be Googled since everybody does this with their docker images.

**Comment by setevoy2** [comment]
@Zciwobukaj I have a blog post [here](https://rtfm.co.ua/firefly-iii-domashnyaya-buxgalteriya/) with detailed description how I did my setup with FF-III, NGINX and Let's Encrypt SSL but it's in Russian. Maybe with Google Translate you can find some helpful info there (but it will break some post formatting), try to check [this link](https://translate.google.ru/translate?sl=ru&tl=en&js=y&prev=_t&hl=en&ie=UTF-8&u=https%3A%2F%2Frtfm.co.ua%2Ffirefly-iii-domashnyaya-buxgalteriya%2F&edit-text=&act=url) (Rus>Eng).
In short - NGINX runs on AWS EC2 instance with Debian as a common service, FF-III as Docker container from Compose file and Let's Encrypt used to obtain SSL certs. MySQL/MariaDB located on a dedicated host.
